source/GameInterface/Services/MobileParties/Patches/UnitRecruitPatch.cs

```diff
@@ -0,0 +1,49 @@
+﻿using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.CampaignBehaviors;
+using TaleWorlds.CampaignSystem.Party;
+using Common.Messaging;
+using GameInterface.Services.MobileParties.Messages;
+using static TaleWorlds.CampaignSystem.CampaignBehaviors.RecruitmentCampaignBehavior;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.Engine;
+using System.Linq;
+
+namespace GameInterface.Services.MobileParties.Patches
+{
+    [HarmonyPatch(typeof(RecruitmentCampaignBehavior))]
+    public class UnitRecruitPatch
+    {
+        [HarmonyPatch("OnUnitRecruited")]
+        public static bool Prefix(CharacterObject troop, int count)
+        {
+            MessageBroker.Instance.Publish(MobileParty.MainParty, new OnUnitRecruited(troop.StringId, count, MobileParty.MainParty.StringId));
+
+            return true;
+        }
+
+        static readonly List<string> args = Utilities.GetFullCommandLineString().Split(' ').ToList();
+
+        [HarmonyPatch("ApplyInternal")] //TODO: Does this fire when other player parties recruit? If so, return false.
```

Comment:
This will also happen for any client party that enters a settlement. You should be able to filter somehow using the `IControlledEntityRegistry` but it needs to be statically passed somehow to this patch. I'll give it some thought

Reply:
It shouldn't happen when you enter a settlement as far as I know.

Reply:
If you think of it on the servers perspective all players are treated as "AI" from the game